ledger-api/grpc-definitions/com/daml/ledger/api/v1/event.proto

```diff
@@ -50,14 +52,30 @@ message CreatedEvent {
   // Required
   Identifier template_id = 3;
 
-  // The key of the created contract, if defined.
+  // The key of the created contract.
+  // This will be set if and only if ``create_arguments`` is set and ``template_id`` defines a contract key.
```

Comment:
is it actually `template_id` defines a contract key rather than actually a `template identified by ``template_id`` ` ?

Reply:
Not quite sure I understand what you're saying. `template_id` is an Identifier that point sto the corresponding template definition.

Reply:
I mean template_id itself does not defines a contract key, it is a template which defines it

Reply:
oh I see, yeah. I'd say you're right technically but we often don't clearly distinguish between the reference to the template via the identifier and the full template definition. Happy to leave it up to you whether you want to change it or not.

Reply:
I think it is clear with some domain knowledge. Have no strong opinion neither.